tools/testinfra/runner/src/main.rs

```diff
@@ -73,45 +88,79 @@ fn main() -> Result<()> {
     // don't run anything when just listing
     if options.list {
         for test in tests {
-            println!("{}", test.name);
+            println!("{}", test_name(&test.target, &test.unit));
         }
         exit(0);
     }
 
+    // connect to DB when it is provided
+    let mut db = match options.conn {
+        None => None,
+        Some(ref uri) => Some(
+            Client::connect(&uri, NoTls)
+                .with_context(|| format!("Couldn't connect to specified test DB at '{}'", uri))?,
+        ),
+    };
+    let disabled = query_disabled(&mut db);
+
     // run tests in parallel (retries share the same thread)
     let total = tests.len();
-    eprintln!("Running {} tests...", total);
+    eprintln!("Found {} tests...", total);
     let _ = rayon::ThreadPoolBuilder::new()
         .num_threads(options.threads)
         .build_global();
     let mut tests: Vec<TestResult> = tests
         .into_par_iter()
         .map(|test| {
-            let test = test.run(options.retries);
+            let run = !disabled.contains(&(test.target.clone(), test.unit.clone()))
+                || options.run_disabled;
+            let test = if run {
```

Comment:
This `let test = if run` is pretty hard to read, a `match` or some kind of `map` construct would be nicer